### ISSUE TITLE
After fresh install and update of the CLI, it reports a new version available for sfdx-cli but will not update it

### ISSUE THREAD

**Issue by RupertBarrow** [forcedotcom/cli#1260]
### Summary

After fresh install and update of the CLI, it reports a new version available for sfdx-cli but will not update it.
Now, when running any SFDX CLI command, a warning is displayed :

### Steps To Reproduce:

0. This issue is independent of the SFDX project
1. Fresh install MacOS Big Sur
2. Fresh install SFDX from the MacOS binary from https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_install_cli.htm
3. Update SFDX CLI with `sfdx update`
4. Now run any SFDX command, eg `sfdx version`

### Expected result

The command should NOT complain about an existing update available for sfdx-cli

### Actual result

% sfdx version
 ›   Warning: sfdx-cli update available from 7.123.0 to 7.123.1.
sfdx-cli/7.123.0 darwin-x64 node-v14.18.1

### System Information

% sfdx version --verbose --json
 ›   Warning: sfdx-cli update available from 7.123.0 to 7.123.1.
{
        "cliVersion": "sfdx-cli/7.123.0",
        "architecture": "darwin-x64",
        "nodeVersion": "node-v14.18.1",
        "pluginVersions": [
                "@oclif/plugin-autocomplete 0.3.0 (core)",
                "@oclif/plugin-commands 1.3.0 (core)",
                "@oclif/plugin-help 3.2.3 (core)",
                "@oclif/plugin-not-found 1.2.4 (core)",
                "@oclif/plugin-plugins 1.10.1 (core)",
                "@oclif/plugin-update 1.5.0 (core)",
                "@oclif/plugin-warn-if-update-available 1.7.0 (core)",
                "@oclif/plugin-which 1.0.3 (core)",
                "@salesforce/sfdx-plugin-lwc-test 0.1.7 (core)",
                "alias 1.1.22 (core)",
                "apex 0.3.0 (core)",
                "auth 1.7.3 (core)",
                "config 1.2.41 (core)",
                "custom-metadata 1.0.12 (core)",
                "data 0.6.3 (core)",
                "dependencies-cli 2.0.1",
                "generator 1.2.0 (core)",
                "limits 1.2.2 (core)",
                "org 1.8.3 (core)",
                "salesforce-alm 53.0.0 (core)",
                "schema 1.0.10 (core)",
                "sfdx-cli 7.123.0 (core)",
                "source 1.2.4 (core)",
                "telemetry 1.2.8 (core)",
                "templates 52.4.0 (core)",
                "trust 1.0.8 (core)",
                "user 1.5.2 (core)"
        ],
        "osVersion": "Darwin 20.6.0"
}

**Comment by RupertBarrow** [comment]
Just literally now, when running the commands to get a screenshot, 7.124.0 came out : all fixed.

Maybe there is a strange behavior when a minor version update (7.123.1) is available for sfdx-cli ? 

<img width="774" alt="image" src="https://user-images.githubusercontent.com/48629751/139400786-48d75fef-1029-4fa6-961b-04f1d512b5c5.png">

**Comment by yippie** [comment]
I came to report this same issue just now. I guess a new version that actually updates is a fix to this.

**Comment by WillieRuemmele** [comment]
so it now correctly updates to `7.124.0`? There was just an issue when you had `7.123.0` installed, and `7.123.1` was `latest`?

**Comment by RupertBarrow** [comment]
Exactly.

**Comment by WillieRuemmele** [comment]
alright, I'll close this as it seems to be fixed in `latest`

**Comment by mikefleming-kaptio-com** [comment]
<img width="630" alt="Screenshot 2021-11-09 at 14 07 26" src="https://user-images.githubusercontent.com/69917488/140938982-bb7a9d9d-f7cb-4647-9855-669bba29a8c0.png">
I'm getting it and it is not on a patch version?

**Comment by douglascayers** [comment]
This "there's an update, but updating doesn't fix it" issue has come back.

I'm on MacOS (intel) and installed the CLI via [link](https://developer.salesforce.com/tools/sfdxcli).

After installing, I run `sfdx update` and `sf update`. Allegedly, I'm on the latest version. However, anytime I run a `sf` command I get the warning that there's an update still available but for the life of me I cannot figure out how to get updated (or get the warning to go away if it's a false positive).

```shell
$ sfdx --version
sfdx-cli/7.194.1 darwin-x64 node-v18.15.0

$ sf --version
 ›   Warning: @salesforce/cli update available from 1.71.0 to 1.72.0.
@salesforce/cli/1.71.0 darwin-x64 node-v18.15.0
```

```shell
$ sfdx update
sfdx-cli: Updating CLI... already on version 7.194.1-faa045e
sfdx-cli: Updating sf... done
# Release notes for 'sfdx':

## 7.194.1 (March 30, 2023) [stable]
...

user plugins are 

$ sf update
@salesforce/cli: Updating CLI... already on version 1.72.0-395c127
@salesforce/cli: Updating plugins... ⢿ [2/4] Fetching packages...
warning "@salesforce/plugin-packaging > @oclif/core > ts-node@10.9.1" has unmet peer dependency "@types/node@*".
@salesforce/cli: Updating plugins... done
```

**Comment by joao-melo-jj2yj8u0** [comment]
I have the same issue using `sfdx update` or `sf update` .  I get:

```
$ sf update
@salesforce/cli: Updating CLI... - fetching manifest
 »   Warning: Running "sf update" has no effect because you're using a version of sf that was installed by sfdx. To
@salesforce/cli: Updating plugins... | [2/4] Fetching packages...
warning "@salesforce/plugin-dev > yeoman-environment@3.15.1" has unmet peer dependency "mem-fs@^1.2.0 || ^2.0.0".
warning "@salesforce/plugin-dev > yeoman-environment@3.15.1" has unmet peer dependency "mem-fs-editor@^8.1.2 || ^9.0.0".
warning "@salesforce/plugin-dev > @oclif/core > ts-node@10.9.1" has unmet peer dependency "@types/node@*".
@salesforce/cli: Updating plugins... done

$ sf dev generate plugin
 »   Warning: @salesforce/cli update available from 1.73.0 to 1.74.7
```

**Comment by Curtis-Bureaux** [comment]
Same issue:

> ❯ sf version
 ›   Warning: @salesforce/cli update available from 1.73.0 to 1.74.7.
@salesforce/cli/1.73.0 darwin-x64 node-v18.15.0
  ~ ························································ at  12:17:58 pm
❯ sf update
@salesforce/cli: Updating CLI... already on version 1.74.7-e719c56
  ~ ························································ at  12:18:07 pm
❯ sfdx version
sfdx-cli/7.197.8 darwin-x64 node-v18.15.0
  ~ ························································ at  12:18:18 pm
❯ sfdx update
sfdx-cli: Updating CLI... already on version 7.197.8-077a68b
sfdx-cli: Updating sf... done
